pyensign/stream.py

```diff
@@ -170,14 +205,18 @@ async def connect(self):
         connection could not be established.
         """
 
-        # Create the gRPC stream from the request iterator
+        # Create the gRPC stream and ensure that it is connected
+        self.stream = self.stub.Publish()
+        await self.stream.wait_for_connection()
+
+        # Send the open stream request to the server
         open_stream = ensign_pb2.PublisherRequest(
             open_stream=ensign_pb2.OpenStream(client_id=self.client_id)
         )
-        stream = self.stub.Publish(RequestIterator(self.queue, open_stream))
+        await self.stream.write(open_stream)
```

Comment:
I'm moving away from the iterator way of interacting with the gRPC API and doing the reads and writes directly.